From 8c1767a02f2a08127d2b66d968be486175c21367 Mon Sep 17 00:00:00 2001
From: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
Date: Fri, 27 Sep 2024 18:57:43 +0200
Subject: [PATCH 78/98] usb: xhci: move Etron PCI IDs to pci_ids.h

The same PCI IDs are needed for the new xhci-pci-etron sub-driver.
Move them to pci_ids.h so they can be re-used instead of having to
copy them over.

Signed-off-by: Martin Blumenstingl <martin.blumenstingl@googlemail.com>

[dts: rebased onto 6.6.144 for 6.6-lts - the local defines here are still
 named PCI_DEVICE_ID_EJ168/EJ188 (not yet ETRON-prefixed) and
 PCI_VENDOR_ID_WCHIC doesn't exist yet in pci_ids.h at this baseline;
 inserted the new defines after PCI_VENDOR_ID_REDHAT (still vendor-ID
 ordered) and updated the two device-ID usages in xhci-pci.c to match]
---
diff --git a/drivers/usb/host/xhci-pci.c b/drivers/usb/host/xhci-pci.c
index 5abc48f14..62ad1d5e8 100644
--- a/drivers/usb/host/xhci-pci.c
+++ b/drivers/usb/host/xhci-pci.c
@@ -34,10 +34,6 @@
 #define PCI_DEVICE_ID_FRESCO_LOGIC_FL1100	0x1100
 #define PCI_DEVICE_ID_FRESCO_LOGIC_FL1400	0x1400
 
-#define PCI_VENDOR_ID_ETRON		0x1b6f
-#define PCI_DEVICE_ID_EJ168		0x7023
-#define PCI_DEVICE_ID_EJ188		0x7052
-
 #define PCI_DEVICE_ID_VIA_VL805			0x3483
 
 #define PCI_DEVICE_ID_INTEL_LYNXPOINT_XHCI		0x8c31
@@ -483,8 +479,8 @@ static void xhci_pci_quirks(struct device *dev, struct xhci_hcd *xhci)
 		xhci->quirks |= XHCI_DEFAULT_PM_RUNTIME_ALLOW;
 
 	if (pdev->vendor == PCI_VENDOR_ID_ETRON &&
-	    (pdev->device == PCI_DEVICE_ID_EJ168 ||
-	     pdev->device == PCI_DEVICE_ID_EJ188)) {
+	    (pdev->device == PCI_DEVICE_ID_ETRON_EJ168 ||
+	     pdev->device == PCI_DEVICE_ID_ETRON_EJ188)) {
 		xhci->quirks |= XHCI_ETRON_HOST;
 		xhci->quirks |= XHCI_RESET_ON_RESUME;
 		xhci->quirks |= XHCI_BROKEN_STREAMS;
diff --git a/include/linux/pci_ids.h b/include/linux/pci_ids.h
index 3dce2be62..3e0d3f482 100644
--- a/include/linux/pci_ids.h
+++ b/include/linux/pci_ids.h
@@ -2588,6 +2588,10 @@
 
 #define PCI_VENDOR_ID_REDHAT		0x1b36
 
+#define PCI_VENDOR_ID_ETRON		0x1b6f
+#define PCI_DEVICE_ID_ETRON_EJ168	0x7023
+#define PCI_DEVICE_ID_ETRON_EJ188	0x7052
+
 #define PCI_VENDOR_ID_SILICOM_DENMARK	0x1c2c
 
 #define PCI_VENDOR_ID_AMAZON_ANNAPURNA_LABS	0x1c36
